sdk/nodejs/runtime/resource.ts

```diff
@@ -519,9 +518,7 @@ async function prepareResource(label: string, res: Resource, custom: boolean, re
 
         // Wait for the parent to complete.
         // If no parent was provided, parent to the root resource.
-        const parentURN = opts.parent
-            ? await opts.parent.urn.promise()
-            : await getRootResource();
+        const parentURN = parent ? await parent.urn.promise() : undefined;
```

Comment:
This `undefined` here. Tracing the code a bit, it seems it will just go straight to gRPC so things like ReadResourceRequest will have parent empty. The engine then assumes it's the `RootResource` so we're all good with this?

Reply:
Yup, this should be fine. getRootResource should always be able to return the stack resource now so this shouldn't be undefined, but if it somehow is the engine will fill it in anyway.